server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1048,17 +1049,26 @@ public void snapshotShard(Store store, MapperService mapperService, SnapshotId s
             final GroupedActionListener<Void> filesListener =
                 new GroupedActionListener<>(allFilesUploadedListener, indexIncrementalFileCount);
             final Executor executor = threadPool.executor(ThreadPool.Names.SNAPSHOT);
+            final AtomicBoolean alreadyFailed = new AtomicBoolean();
             for (BlobStoreIndexShardSnapshot.FileInfo snapshotFileInfo : filesToSnapshot) {
                 executor.execute(new ActionRunnable<>(filesListener) {
                     @Override
                     protected void doRun() {
                         try {
-                            snapshotFile(snapshotFileInfo, indexId, shardId, snapshotId, snapshotStatus, store);
+                            if (alreadyFailed.get() == false) {
+                                snapshotFile(snapshotFileInfo, indexId, shardId, snapshotId, snapshotStatus, store);
+                            }
```

Comment:
Should we call filesListener.onFailure() with a IndexShardSnapshotFailedException and an appropriate comment in case the upload is skipped because of the `alreadyFailed` flag?

Reply:
If we do that, we get all those execeptions added to the overall exception again, precisely what I tried avoiding here. Note that you could theoretically have a pretty large number of files here.

Reply:
Right - then at least a trace log would be helpful?

Reply:
To be honest, I can't see what it would be useful for?
We'll get some indication that things were aborted anyway from the files that were actually aborted to set the flag and we will not update the blob `index-N` in the blob as a result of failing here as well -> seems like the information on what precise files were never uploaded is completely irrelevant?

Reply:
OK

(Monday morning reviews, you know)